grpc-gateway/service/updateResource_test.go

```diff
@@ -272,3 +273,70 @@ func TestRequestHandlerGetAfterUpdateResource(t *testing.T) {
 	endData := getData(deviceID)
 	require.Equal(t, startData, endData)
 }
+
+func TestRequestHandlerRunMultipleUpdateResource(t *testing.T) {
+	deviceID := test.MustFindDeviceByName(test.TestDeviceName)
+	numIteration := 50
+	ctx, cancel := context.WithTimeout(context.Background(), config.TEST_TIMEOUT+(time.Second*2*time.Duration(numIteration)))
+	defer cancel()
+
+	raCfg := raTest.MakeConfig(t)
+	raCfg.Clients.Eventstore.SnapshotThreshold = 5
+	tearDown := service.SetUp(ctx, t, service.WithRAConfig(raCfg))
+	defer tearDown()
+	ctx = kitNetGrpc.CtxWithToken(ctx, oauthTest.GetDefaultAccessToken(t))
+
+	conn, err := grpc.Dial(config.GRPC_HOST, grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{
+		RootCAs: test.GetRootCertificatePool(t),
+	})))
+	require.NoError(t, err)
+	defer func() {
+		_ = conn.Close()
+	}()
+	c := pb.NewGrpcGatewayClient(conn)
+
+	resources := test.GetAllBackendResourceLinks()
+	_, shutdownDevSim := test.OnboardDevSim(ctx, t, c, deviceID, config.GW_HOST, resources)
+	defer shutdownDevSim()
+
+	for i := 0; i < numIteration; i++ {
+		t.Logf("TestRequestHandlerMultipleUpdateResource:run %v\n", i)
+		lightHref := test.TestResourceLightInstanceHref("1")
+		for i := 1; i >= 0; i-- {
```

Comment:
maybe a different indexing variable would be more readable, it's weird to see that both outer and inner use `i`.

Reply:
fixed